x-pack/plugins/ml/public/application/memory_usage/memory_tree_map/tree_map.tsx

```diff
@@ -0,0 +1,188 @@
+import { Chart, Settings, Partition, PartitionLayout, ShapeTreeNode } from '@elastic/charts';
+import { FIELD_FORMAT_IDS } from '@kbn/field-formats-plugin/common';
+import {
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiEmptyPrompt,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { MemoryUsageInfo } from '../../../../common/types/trained_models';
+import { JobType, MlSavedObjectType } from '../../../../common/types/saved_objects';
+import { useTrainedModelsApiService } from '../../services/ml_api_service/trained_models';
+import { LoadingWrapper } from '../../jobs/new_job/pages/components/charts/loading_wrapper';
+import { useFieldFormatter } from '../../contexts/kibana';
+
+import { useRefresh } from '../../routing/use_refresh';
+import { getMemoryItemColor } from '../memory_item_colors';
+
+interface Props {
+  node?: string;
+  type?: MlSavedObjectType;
+  height?: string;
+}
+
+const DEFAULT_CHART_HEIGHT = '400px';
+
+const TYPE_OPTIONS: TreeMapOptions[] = [
+  {
+    label: i18n.translate('xpack.ml.memoryUsage.treeMap.adLabel', {
+      defaultMessage: 'Anomaly detection jobs',
+    }),
+    objectType: 'anomaly-detector',
+    color: getMemoryItemColor('anomaly-detector'),
+  },
+  {
+    label: i18n.translate('xpack.ml.memoryUsage.treeMap.dfaLabel', {
+      defaultMessage: 'Data frame analytics jobs',
+    }),
+    objectType: 'data-frame-analytics',
+    color: getMemoryItemColor('data-frame-analytics'),
+  },
+  {
+    label: i18n.translate('xpack.ml.memoryUsage.treeMap.modelsLabel', {
+      defaultMessage: 'Trained models',
+    }),
+    objectType: 'trained-model',
+    color: getMemoryItemColor('trained-model'),
+  },
+];
+
+type TreeMapOptions = EuiComboBoxOptionOption & { objectType: MlSavedObjectType };
+
+export const JobMemoryTreeMap: FC<Props> = ({ node, type, height }) => {
+  const bytesFormatter = useFieldFormatter(FIELD_FORMAT_IDS.BYTES);
+  const refresh = useRefresh();
+  const chartHeight = height ?? DEFAULT_CHART_HEIGHT;
+
+  const trainedModelsApiService = useTrainedModelsApiService();
+  const [allData, setAllData] = useState<MemoryUsageInfo[]>([]);
+  const [data, setData] = useState<MemoryUsageInfo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [selectedOptions, setSelectedOptions] = useState<TreeMapOptions[]>(TYPE_OPTIONS);
+
+  const filterData = useCallback(
+    (dataIn: MemoryUsageInfo[]) => {
+      const labels = selectedOptions.map((o) => o.objectType);
+      return dataIn.filter((d) => labels.includes(d.type));
+    },
+    [selectedOptions]
+  );
+
+  const onTypeChange = useCallback((types: EuiComboBoxOptionOption[]) => {
+    setSelectedOptions(types as TreeMapOptions[]);
+  }, []);
+
+  const loadJobMemorySize = useCallback(async () => {
+    setLoading(true);
+    const resp = await trainedModelsApiService.memoryUsage(type, node);
```

Comment:
should it be wrapped in `try...catch`? and perhaps showing an error toast in case of a failure?

Reply:
Updated in https://github.com/elastic/kibana/pull/149419/commits/882e0dd38a432a6b215884289b86edc84a35a7a0